sympy/core/mul.py

```diff
@@ -1880,7 +1880,8 @@ def coeff_exp(term, x):
         try:
             for t in self.args:
                 coeff, exp = t.leadterm(x)
-                if not coeff.has(x):
+                if not (coeff.has(x) or coeff.is_Symbol) \
```

Comment:
Why cannot `coeff` be a `Symbol`, a constant, for example?

Reply:
Yeah, I think this condition doesn't make sense. I was trying to resolve the other simplification issue, but I don't think this is the right place to resolve it. It is also getting very difficult to debug all of this together. I guess I will split the work here into 2 PRs.